backend/day-15/day-15-challenge.py

```diff
@@ -0,0 +1,181 @@
+import json
+import validate
+
+class PasswordManager:
+    def __init__(self, website: str, email: str, password: str):
+        self.website = website
+        self.email = email
+        self.password = password
+
+    def __loadData(self, path: str):
+        with open(path, 'r') as file:
+            return json.load(file)
+    
+    def __updateData(self, path: str, data: object):
+        with open(path, 'w') as file:
+            json.dump(data, file, indent=4)
+
+    def addData(self):
+        try:
+            data = self.__loadData('backend/day-15/data.json')
+
+            new_data = {
+                "email": self.email,
+                "password": self.password
+            }
+
+            website_key = self.getWebsiteKey()
+            if website_key in data and new_data in data[website_key]:
+                print(f"Password already exists for website: {self.getWebsiteName(website_key)}")
+                return
+            else:
+                data.setdefault(website_key, []).append(new_data)
+
+            self.__updateData('backend/day-15/data.json', data)
+            print("Password added successfully!")
+        except Exception as e:
+            print(f"An error occured: {e}")
+
+    def getData(self, filter_key = None):
+        try: 
+            data = self.__loadData('backend/day-15/data.json')
+            filtered_items = {}
+            if filter_key:
+                if filter_key in data:
+                    filtered_items = {filter_key: data[filter_key]}
+                else:
+                    for website_key, accounts in data.items():
+                        filtered_accounts= [
+                            account for account in accounts if filter_key in account.values()
+                        ]
+                        if filtered_accounts:
+                            filtered_items[website_key] = filtered_accounts
+                if filtered_items:
+                    self.displayData(filtered_items)
+                else:
+                    print("Keyword is not found on our record!")
+            else:
+                self.displayData(data)
+        except Exception as e:
+            print(f"An error occured: {e}")
+    
+    def displayData(self, data: object):
+        for website, account in data.items():
+            print(f"Website: {self.getWebsiteName(website)}")
+            for details in account:
+                print(f"\tEmail: {details['email']}")
+                print(f"\tPassword: {details['password']}")
+
+    def deleteData(self):
+        try:
+            data = self.__loadData('backend/day-15/data.json')
+            website_key = self.getWebsiteKey()
+            for account in data.get(website_key, []):
+                if account.get('email') == self.email and account.get('password') == self.password:
```

Comment:
Might be a good idea to put this condition logic to a method in your password manager class as this is also repeated on line90